torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -0,0 +1,401 @@
+#include <torch/csrc/distributed/rpc/tensorpipe_agent.h>
+
+#include <torch/csrc/distributed/rpc/request_callback_impl.h>
+#include <torch/csrc/distributed/rpc/utils.h>
+
+#include <tensorpipe/channel/basic/context.h>
+#ifdef TP_ENABLE_CMA
+#include <tensorpipe/channel/cma/context.h>
+#endif
+#ifdef TP_ENABLE_SHM
+#include <tensorpipe/transport/shm/context.h>
+#endif
+#include <tensorpipe/transport/uv/context.h>
+
+namespace torch {
+namespace distributed {
+namespace rpc {
+
+using clock = std::chrono::high_resolution_clock;
+using nss = std::chrono::nanoseconds;
+
+std::atomic<uint32_t> TensorPipeAgent::nextShmAddressID_(0);
+
+TensorPipeAgent::TensorPipeAgent(
+    worker_id_t selfId,
+    std::string selfName,
+    std::shared_ptr<::c10d::Store> addressStore,
+    TensorPipeRpcBackendOptions opts)
+    : RpcAgent(
+          WorkerInfo(std::move(selfName), std::move(selfId)),
+          std::make_unique<RequestCallbackImpl>(),
+          opts.rpcTimeout),
```

Comment:
I'm guessing we don't honor the rpcTimeout in TensorPipe yet? Can we add a TODO for this?